engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java

```diff
@@ -233,6 +234,37 @@ public void replaceBlock(
 
     }
 
+    @Command(shortDescription = "Add multiple blocks to inventory",
+            helpText = "Adds all blocks that match the search parameter into your inventory",
+            runOnServer = true, requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String bulkGiveBlock(
+            @Sender EntityRef sender,
+            @CommandParam("searched") String searched) {
+        String searchLowercase = searched.toLowerCase();
+        List<String> blocks = findBlockMatches(searchLowercase);
+        String result = "Found " + blocks.size() + " block matches when searching for '" + searched + "'.";
+        if (blocks.size() > 0) {
+            result += "\nBlocks:";
+            for (String block : blocks) {
+                result += "\n" + block;
+                giveBlock(sender, block, null, null);
```

Comment:
I think it is probably overkill, but that's fourth parameter right there, the second null. You can check out how it works by simply playing with giveBlock :-)

`giveblock stone 99 stair`

Should give you 99 stone stair-shaped blocks and is an example from the Playing doc ... although after fixing it as the doc has the quantity and shape switched ;-) Fixing next time I push something.